### DISCUSSION TITLE
The fees in the Sepolia test net are too high, is this normal?

### DISCUSSION THREAD

**Discussion by gsangeryee** [Cyfrin/foundry-full-course-cu#2567] [Q&A] [answered]
The fees in the Sepolia testnet and ZkSync Sepolia testnet are too high.

First of all,

When I was following the tutorial "12. Deploying your first contract" and trying to deploy to the Sepolia testnet using Injected Provider - MetaMask, I encountered the following error:

```
Gas estimation failed.
Returned error: contract creation code storage ouf of gas.
```

<img width="1162" alt="Screenshot 2024-09-28 at 09 24 57" src="https://github.com/user-attachments/assets/3ad233d0-e5b2-4c4d-807a-cefd4c2e84fa">

I copied my code from https://github.com/Cyfrin/remix-storage-factory-cu/blob/main/SimpleStorage.sol.

My MetaMask account has 0.07 SepoliaETH.

I don't think it's a code issue because I tried deploying the code to the Zksync Sepolia testnet, hoping to reduce the gas fee.

However, when I bridged SepoliaETH to the Zksync Sepolia testnet using the Zksync bridge, I found that the fee was also very high. Please see the screenshot:

![zksync](https://github.com/user-attachments/assets/cb5a5945-0eac-4822-923c-b700dc014678)

Fee: 0.0360496

<img width="1070" alt="Screenshot 2024-09-28 at 09 29 22" src="https://github.com/user-attachments/assets/6fab36c4-c8ed-4932-98c0-5fb2d8804982">

Fee: 0.02859436

When I checked the transactions on the Sepolia testnet using explore, I found that their fees were very low.

<img width="1168" alt="Screenshot 2024-09-28 at 09 49 20" src="https://github.com/user-attachments/assets/bf565a4a-4be9-463c-8401-390f229f4279">

Fee: 0.0076798 

So my question, is such a high fee normal?

**Comment by EngrPips** [comment]
Hello @gsangeryee, I have seen many people complain the same way, so it made me start to wonder if it is a congestion issue on the Sepolia network, but I am not sure. I deployed `SimpleStorage` contract yesterday, and the gas fee was around 0.1 ether for me.

**Reply by ianwright27** [reply]
Me too, use Layer 2 testnets, thank me later. 
(That tutorial was from around 2022, activity on Layer 1 was minimal then compared to now)

**Reply by EngrPips** [reply]
Interesting

**Comment by gsangeryee** [comment]
Hi @EngrPips , But why are the fees for the transactions recently,  I queried on the Sepolia Testnet 
 and ZkSync Sepolia Testnet so low?  Confused!

**Reply by EngrPips** [reply, marked as answer]
You know, blockchain transaction fee is not something fixed at all. It rises and falls due to the demand and supply of the block space on the blockchain, So it makes sense that sometimes the fee is so high and sometimes the fee is very low.

**Reply by gsangeryee** [reply]
Got it.  Thanks. @EngrPips 

I guess I'll have to get more SpeoliaETH before I can test further (Claim SepoliaETH before Study 👍  ). It would be ideal to do everything on ZKSync Speolia, but the bridge fees are pretty steep right now. Got it.

But the current transaction fee for bridging SepoliaETH from the Sepolia to the ZKSync Sepolia is also high.

<img width="1218" alt="Screenshot 2024-09-29 at 15 41 49" src="https://github.com/user-attachments/assets/271ddf9a-857f-483b-9378-42ed3c0065e0">

Recommend some good facuets:

SepoliaETH:
- https://cloud.google.com/application/web3/faucet/ethereum/sepolia (0.05ETH / 24hours)
- https://faucets.chain.link/sepolia (0.1ETH / 24hours)

ZKSync SepoliaETH
- https://thirdweb.com/zksync-sepolia-testnet (0.01 ETH / 24 hours)

**Reply by EngrPips** [reply]
I always use the chainlink faucet myself. I think working with `sepolia` is the best because working with `zksync-sepolia` is not very smooth at the moment.

**Comment by sorXCode** [comment]
You can check the gas price estimation on Sepolia using https://sepolia.beaconcha.in/gasnow. The lower the gas price, the lesser the transaction fee.

**Reply by ianwright27** [reply]
How do you check for it here, there's a lot of information, it's not really clear.

**Comment by EngrPips** [comment]
Sepolia Gas fee is outrageous lately, I must say.

**Reply by ianwright27** [reply]
Tell me about it

**Reply by EngrPips** [reply]
Why not my Friend? https://sepolia.beaconcha.in/gasnow

**Reply by ianwright27** [reply]
I found a work around, I added Base Sepolia network to my MetaMask instead and deployed from Remix with that instead. Layer 2 is 🚀✨ So fast and very cheap on the gas fees.

**Reply by EngrPips** [reply]
You are a giga chad sir!
